src/analyzing/AST_to_IL.ml

```diff
@@ -1241,6 +1241,9 @@ and stmt_aux env st =
       let ss, e' = expr_with_pre_stmts env e in
       let lv = lval_of_ent env ent in
       ss @ [ mk_s (Instr (mk_i (Assign (lv, e')) (Related (G.S st)))) ]
+  | G.DefStmt (_, G.ClassDef { cbody = fields; _ }) ->
```

Comment:
In general I don't think this is correct, the fields could be modified by calls to methods no? It's correct if the only thing you're analyzing are the fields themselves. It would be useful to get more context here.

Reply:
Certainly, it is possible to modify the fields after initialization. However, the primary objective of this PR is to perform data flow analysis specifically during object initialization. The purpose is to address issue #7742 by resolving a limitation that cannot be resolved using AST alone. In the AST, each field in the object definition is treated independently, but in taint mode, there is a partial feasibility in handling this scenario.

However, since I lack sufficient context on the design decisions made during IL transformation, I am also open to discarding these changes if the negative side effects outweigh the positives.

Reply:
Then IMO we should have a special visiting step that just analyzes fields in `Match_tainting_mode.ml`. Although I think this could be solved already with sym-prop. The #7742's case should be even simpler because it only needs const-prop. However when I open that issue and I follow the Playground link, what I see is a search-mode rule.

Reply:
The author of #7742 attempted to perform semi-taint analysis in search mode. However, this approach would not work with class fields without significant modifications to the class_definition AST. The PR enables the same analysis in taint mode, which appears to be more suitable for this particular case.

Reply:
I'm not so sure about that, have you actually talked to Lewis about it?

Reply:
I am happy to merge this, I am not so sure you are actually fixing #7742.